assets/section-main-product.css

```diff
@@ -1178,3 +1195,65 @@ a.product__text {
   display: block;
   padding-bottom: 100%;
 }
+
+/* collapsed grid overrides */
+@media screen and (min-width: 750px) {
+  .product--stacked .grid--collapsed-y .grid__item:first-child {
+    --top-left-radius: var(--border-radius);
+    --top-right-radius: var(--border-radius);
+  }
+
+  .product--thumbnail .grid--collapsed .is-active,
+  .product--thumbnail_slider .grid--collapsed .is-active {
+    --top-left-radius: var(--border-radius);
+    --top-right-radius: var(--border-radius);
+    --bottom-right-radius: var(--border-radius);
+    --bottom-left-radius: var(--border-radius);
+  }
+}
+
+@media screen and (min-width: 750px) and (max-width: 989px) {
+  .product--stacked .grid--collapsed-y .grid__item:last-child,
+  .product--stacked .grid--collapsed-x:not(.grid--collapsed-y) .grid__item {
+    --bottom-right-radius: var(--border-radius);
+    --bottom-left-radius: var(--border-radius);
+  }
+
+  .product--stacked .grid--collapsed-x:not(.grid--collapsed-y) .grid__item {
+    --top-left-radius: var(--border-radius);
+    --top-right-radius: var(--border-radius);
+  }
+}
+
+@media screen and (min-width: 990px) {
+  .product--stacked .grid--collapsed-y .grid__item:first-child {
+    --top-left-radius: var(--border-radius);
+    --top-right-radius: var(--border-radius);
+  }
+
+  .product--stacked .grid--collapsed-y .grid__item:only-child,
+  .product--stacked .grid--collapsed-y .grid__item:last-child:nth-child(even) {
+    --bottom-left-radius: var(--border-radius);
+    --bottom-right-radius: var(--border-radius);
+  }
+
+  .product--stacked .grid--collapsed-y .grid__item:last-child:nth-child(odd) {
+    --bottom-right-radius: var(--border-radius);
+  }
+
+  .product--stacked .grid--collapsed-y .grid__item:nth-last-child(2):nth-child(even):not(.product__media-item--full) {
+    --bottom-left-radius: var(--border-radius);
+  }
+
+  .product--stacked .grid--collapsed-x:not(.grid--collapsed-y) .grid__item:first-child,
+  .product--stacked .grid--collapsed-x:not(.grid--collapsed-y) .grid__item:nth-child(even) {
+    --top-left-radius: var(--border-radius);
+    --bottom-left-radius: var(--border-radius);
+  }
+
+  .product--stacked .grid--collapsed-x:not(.grid--collapsed-y) .grid__item:nth-child(odd),
+  .product--stacked .grid--collapsed-x:not(.grid--collapsed-y) .grid__item:last-child:nth-child(even) {
+    --top-right-radius: var(--border-radius);
+    --bottom-right-radius: var(--border-radius);
+  }
+}
```

Comment:
Not added in here but I guess coming from main maybe ? There is some padding right applied to the first product image which messes with zero grid space: ![](https://screenshot.click/22-52-1p1g1-168on.png)

Reply:
That actually turned in to quite a thing haha. We apparently have a case where there may be only one _visible_ media item but it isn't the only `<li>` child so we can't target it with CSS alone. Separately, but also related, is even when we have this case, we still render the slider classes for some reason, which means it inherits all the slider related margins, widths, and of course collapsed radius values.. This section really needs a cleanup. 

I think this is safe to include here, but let me know what you think. The i_deal_ solution to all this is a much larger task.
https://github.com/Shopify/dawn/pull/1478/commits/678cb77372e81d75e83dc1a1ecc10bfc70be9705

Reply:
The issue you screenshot is because that variant:first-child rule with the padding actually needed be referring to the "only visible media" item.